crates/idxdb-store/src/transaction/mod.rs

```diff
@@ -95,7 +95,8 @@ impl WebStore {
             .get_account(delta.id())
             .await?
             .ok_or(StoreError::AccountDataNotFound(delta.id()))?
-            .into();
+            .try_into()
+            .map_err(|_| StoreError::AccountDataNotFound(delta.id()))?;
```

Comment:
Would it make sense to expand `AccountDataNotFound` to optionally pass in more context? Maybe something like 

```
type Context = String;
#[error("account data wasn't found for account id {0}:{1}")]
AccountDataNotFound(AccountId, Option<Context>)
```. 

And then,
```suggestion
            .map_err(|client_err| StoreError::AccountDataNotFound(delta.id(), client_err))?;
```

I think, when possible, providing as much context as possible (within reason) aids debugging later on. Having said that, maybe it's a bit overkill, so just a suggestion.